conformance/scenarios/httpgateway/httpgateway_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package httpgateway
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	tstate "sigs.k8s.io/gateway-api/conformance/state"
+	"sigs.k8s.io/gateway-api/conformance/utils"
+)
+
+var (
+	state *tstate.Scenario
+)
+
+func TestHTTPGateway(t *testing.T) {
```

Comment:
@bowei @hbagdi Thanks for the feedback on these tests earlier this week! I ended up moving back to an approach that didn't rely on godog, shown here. The structure is highly similar to the features file the first file in this PR. These tests fully pass with Istio, but of course this is a tiny subset of what we'd want to include in a full conformance suite. 

Some of the key advantages:
- Meaningful output as tests are running
- No additional translation/abstraction layer
- Simpler debugging
- Less dependencies
- More flexibility when writing complex tests

Of course we also lose the following:
- Feature files that are easy to read
- Pretty output that comes with godog
- Consistency with ingress-controller-conformance

This still needs work in terms of cleanup and restructuring, but I'm interested in high level feedback on how these tests are actually structured. Does it feel simple enough to write and understand?